### PR TITLE
+ add wsl2 support

### PR DESCRIPTION
**Pull request by BacLuc**

### CODE REVIEW

File: docs/wsl2.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+WSL 2
+=======
+
+WSL 2 bietet die Möglichkeit, einen Linux-Kernel parallel zu Windows auszuführen.
+Der Zugriff auf das Windows-Dateisystem ist möglich, jedoch etwas langsam.
+Deshalb sollte man seine Dateien im WSL 2 Filesystem bearbeiten.
+Mit den neuesten Versionen von Windows 10 und Windows 11 können auch
+GUI-Anwendungen ausgeführt werden.
+Es wird empfohlen, eine separate Ubuntu WSL 2 Instanz für den Start zu erstellen,
```
Comment (BacLuc) [inline docs/wsl2.md:9]:
```suggestion
Es wird empfohlen, eine separate Ubuntu WSL 2 Instanz für den Jumpstart zu erstellen,
```

File: ansible/playbook.yml [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
+      ignore_errors: true
+      changed_when: true
+      register: wsl_is_wsl_command
+    - name: Set facts if running in wsl
+      ansible.builtin.set_fact:
+        wsl_is_wsl: '{{ true if (wsl_is_wsl_command.rc == 0) else false }}'
     - ansible.builtin.include_tasks: tasks/ubuntu_apt.yml
     - ansible.builtin.include_tasks: tasks/ubuntu_desktop.yml
     - ansible.builtin.include_tasks: tasks/virtualbox_additions.yml
+    - ansible.builtin.include_tasks: tasks/virtualbox_additions.yml
```
Comment (langchr86) [inline ansible/playbook.yml:15]: Duplicated line

File: ansible/playbook.yml [hunk 3]
```diff
@@ -1,9 +1,18 @@
...
 
   pre_tasks:
+    - name: Check if running in wsl 2
+      ansible.builtin.shell: wsl.exe -- ls
+      ignore_errors: true
+      changed_when: true
+      register: wsl_is_wsl_command
+    - name: Set facts if running in wsl
+      ansible.builtin.set_fact:
+        wsl_is_wsl: '{{ true if (wsl_is_wsl_command.rc == 0) else false }}'
```
Comment (langchr86) [inline ansible/playbook.yml:11]: Move this whole block into a separate tasks file similar to the below ones.

File: ansible/tasks/ubuntu_desktop.yml [hunk 4]
```diff
@@ -9,29 +9,34 @@
   ansible.builtin.apt:
     name: python3-software-properties
     state: absent
-  when: ubuntu_desktop.rc == 1
+  when: (ubuntu_desktop.rc == 1) and (not wsl_is_wsl)
```
Comment (langchr86) [inline ansible/tasks/ubuntu_desktop.yml:12]: Do not change all these included task files but apply the `when` conditional to the place where those files are included.

**Review by langchr86 — CHANGES_REQUESTED** [review]

File: docs/wsl2.md (hunk 1 above)
Comment (BacLuc) [reply docs/wsl2.md]: done

File: ansible/playbook.yml (hunk 3 above)
Comment (BacLuc) [reply ansible/playbook.yml]: done

File: ansible/tasks/ubuntu_desktop.yml (hunk 4 above)
Comment (BacLuc) [reply ansible/tasks/ubuntu_desktop.yml]: done

**Comment by langchr86** [comment]
Looks good to me. I enabled the vagrant-up action for pull-requests now. Can you somehow re-push the PR or trigger the action manually to see if the ansible still works?

**Review by langchr86 — APPROVED** [review]

**Comment by BacLuc** [comment]
Not needed anymore, we now have github codespaces

**Pull request closed without merging**